src/ja/firecross/src/eu/kanade/tachiyomi/extension/ja/firecross/FireCross.kt

```diff
@@ -0,0 +1,119 @@
+package eu.kanade.tachiyomi.extension.ja.firecross
+
+import eu.kanade.tachiyomi.multisrc.clipstudioreader.ClipStudioReader
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.asObservable
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import keiyoushi.utils.toJsonString
+import keiyoushi.utils.tryParse
+import okhttp3.FormBody
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class FireCross : ClipStudioReader(
+    "FireCross",
+    "https://firecross.jp",
+    "ja",
+) {
+    private val apiUrl = "https://firecross.jp/api"
+    private val dateFormat = SimpleDateFormat("yyyy/M/d", Locale.JAPAN)
+
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/ebook/comics?sort=1&page=$page", headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select("ul.seriesList li.seriesList_item").map { element ->
+            SManga.create().apply {
+                element.selectFirst("a.seriesList_itemTitle")!!.let { a ->
+                    setUrlWithoutDomain(a.attr("href"))
```

Comment:
```suggestion
                    setUrlWithoutDomain(a.absUrl("href"))
```

Suggestion applies throughout this changeset.